app/models/assignment.rb

```diff
@@ -79,11 +79,15 @@ def send_reminders!
   def overlaps_any?
     overlapping_assignments = if new_record?
                                 roster.assignments
-                                    .where("start_date <= ? AND end_date >= ?",
-                                           end_date, start_date)
+                                      .where('
+                                      start_date <= ? AND end_date >= ?',
+                                             end_date, start_date)
                               else
-                                roster.assignments.where("start_date <= ? AND end_date >= ? AND id != ?",
-                                           end_date, start_date, id)
+                                roster.assignments
+                                      .where('
```

Comment:
These breaks are in weird places.  The reason to do something different depending on `new_record?` is because we want to avoid comparing to `NULL`, which is undefined. Is it more or less bad to do any of these:

```ruby
overlapping_assignments = roster.assignments.where(
  "start_date <= ? AND end_date >= ? AND id != ?",
  end_date, start_date, id.to_i
)
# OR
overlapping_assignments = roster.assignments.where(
  "start_date <= ? AND end_date >= ? AND id != ?",
  end_date, start_date, (persisted? && id)
)
```

Hacky?

Reply:
I see what you mean, I like option #2

Reply:
But what if the assignment is new and the id isn't yet persisted?

Reply:
Those are the same thing. `persisted?` is the opposite of `new_record?`

Reply:
I had the wrong impression of what would be returned by the combination of `(persisted? && id)` if id was null. I think I'll go with what you first suggested since that seemed a little more obvious to me